glue/sample/src/sinter/_decoding_pymatching.py

```diff
@@ -10,12 +19,15 @@ def decode_shots_bit_packed(
             *,
             bit_packed_detection_event_data: 'np.ndarray',
     ) -> 'np.ndarray':
-        return self.matcher.decode_batch(
+        result = self.matcher.decode_batch(
             shots=bit_packed_detection_event_data,
             bit_packed_shots=True,
             bit_packed_predictions=True,
             return_weights=False,
         )
+        if isinstance(result, tuple):
+          return result[0]  # pymatching returned predictions and weights
```

Comment:
If this ever actually executed, it would be a bug. Raise an exception, or do a cast.